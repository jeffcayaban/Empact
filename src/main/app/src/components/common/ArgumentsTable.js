import React from 'react'
import {Table} from "react-bootstrap";

import TableFooter from "./TableFooter";
import ArgumentEntry from "./ArgumentEntry";
import { withRouter } from "react-router-dom";
import {
    getArgumentsByArgumentId,
    getArgumentsByRootPetitionId, getMostDiscussedArgumentsForArgument,
    getMostDiscussedArgumentsForPetition
} from "../../utils/AppUtils";
import {
    ARGUMENT, ASCENDING,
    DESCENDING,
    DISPLAY_SORT_BY_VALUES, LEAST_DISCUSSED, MOST_DISCUSSED,
    NEWEST,
    PETITION,
    SORT_BY_VALUES
} from "../../constants";
import TableSortByBtn from "./TableSortByBtn";
import NoContentAlert from "./alerts/NoContentAlert";

/**
 * Displays a set of arguments and provides pagination and sorting options.
 */

class ArgumentsTable extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            currentSortByDisplay: NEWEST,

            argumentsList: [],
            currentSort: DESCENDING,
            pageNumber: 0,
            pageSize: 5,
            totalPages: 0,
            argumentsLast: true,
            byNoOfChildArgs: false
        }
    }

    // Fetches arguments from the server.
    getArgumentsFromServer(page, size, sort, subjectId, isSupporting, parentContentType, byNoOfChildArgs) {
        if (parentContentType === PETITION) {
            // Assumes the parent content is a Petition.

            if (!byNoOfChildArgs) {
                // Fetches the arguments in order of creation date.

                getArgumentsByRootPetitionId(page, size, sort, subjectId, isSupporting)
                    .then(response => {
                        this.setState({
                            argumentsList: response.content,
                            pageNumber: response.page,
                            totalPages: response.totalPages,
                            argumentsLast: response.last,
                            currentSort: sort,
                            byNoOfChildArgs: false
                        })
                    })
                    .catch(error => console.error(error));
            } else {
                // Fetches the arguments by the number of arguments to it.

                getMostDiscussedArgumentsForPetition(page, size, sort, subjectId, isSupporting)
                    .then(response => {
                        this.setState({
                            argumentsList: response.content,
                            pageNumber: response.page,
                            totalPages: response.totalPages,
                            argumentsLast: response.last,
                            currentSort: sort,
                            byNoOfChildArgs: true
                        })
                    })
                    .catch(error => console.error(error));
            }
        } else {
            // Assumes the parent content is an Argument.

            if (!byNoOfChildArgs) {
                // Fetches the arguments in order of creation date.

                getArgumentsByArgumentId(page, size, sort, subjectId, isSupporting)
                    .then(response => {
                        this.setState({
                            argumentsList: response.content,
                            pageNumber: response.page,
                            totalPages: response.totalPages,
                            argumentsLast: response.last,
                            byNoOfChildArgs: false
                        })
                    })
                    .catch(error => console.error(error));
            } else {
                // Fetches the arguments by the number of arguments to it.

                getMostDiscussedArgumentsForArgument(page, size, sort, subjectId, isSupporting)
                    .then(response => {
                        this.setState({
                            argumentsList: response.content,
                            pageNumber: response.page,
                            totalPages: response.totalPages,
                            argumentsLast: response.last,
                            currentSort: sort,
                            byNoOfChildArgs: true
                        })
                    })
                    .catch(error => console.error(error));
            }
        }
    }

    // Calls the method to get arguments from the server.
    handleLoadArguments(pageNumber, currentSort = this.state.currentSort, byNoOfChildArgs = this.state.byNoOfChildArgs) {
        const { subjectId, isSupporting, parentContentType } = this.props;

        this.getArgumentsFromServer(
            pageNumber,
            this.state.pageSize,
            currentSort,
            subjectId,
            isSupporting,
            parentContentType,
            byNoOfChildArgs
        );
    }

    // Handles the event of where the arguments are to be sorted by a given parameter.
    handleChangeSortBy(key) {
        const { currentSortByDisplay } = this.state;

        const newSortByValue = SORT_BY_VALUES[key - 1];
        const newCurrentSortByDisplay = DISPLAY_SORT_BY_VALUES[key - 1];

        if (newCurrentSortByDisplay !== currentSortByDisplay) {
            // Assumes that the sort by option has been changed.

            if (key === "1" || key === "2") {
                // Assumes that the arguments are to be sorted by creation date.

                this.setState({ currentSortByDisplay: newCurrentSortByDisplay, byNoOfChildArgs: false });
                this.handleLoadArguments(0, newSortByValue, false);
            } else {
                // Assumes that the arguments are to be sorted by the number of arguments to it.

                if (newCurrentSortByDisplay === MOST_DISCUSSED) {
                    // Assumes that arguments are to be sorted with the most discussed arguments first.

                    this.setState({ currentSortByDisplay: newCurrentSortByDisplay, byNoOfChildArgs: true });
                    this.handleLoadArguments(0, DESCENDING, true);
                } else if (newCurrentSortByDisplay === LEAST_DISCUSSED) {
                    // Assumes that arguments are to be sorted with the least discussed arguments first.

                    this.setState({ currentSortByDisplay: newCurrentSortByDisplay, byNoOfChildArgs: true });
                    this.handleLoadArguments(0, ASCENDING, true);
                }
            }
        }
    }

    componentDidMount() {
        this.handleLoadArguments(this.state.pageNumber);
    }

    render() {
        const { argumentsList, pageNumber, totalPages, currentSortByDisplay } = this.state;
        const { parentArgument, isSupporting, parentContentCreatorId, parentContentType } = this.props;

        if (argumentsList.length === 0) {
            // Assumes that there are no arguments to be displayed so an error is displayed instead.
            return <NoContentAlert contentType={ARGUMENT} />;
        } else {
            // Assumes that there are arguments to be displayed.
            return (
                <div>
                    <div className={"argumentsTableHeader"}>
                        <div className={"spaceBetweenElements"} style={{ paddingTop: '7px', paddingBottom: '7px' }}>
                            <TableSortByBtn
                                id={`${isSupporting}-args-sortby`}
                                title={currentSortByDisplay}
                                onSelect={this.handleChangeSortBy.bind(this)}
                                totalContentCount={this.props.totalNoOfArgs}
                                showMoreOptions={true}
                            />
                        </div>
                    </div>

                    <Table responsive hover className={"argumentsTable"}>
                        <tbody>
                            { argumentsList.map((argument, index) =>
                                <ArgumentEntry
                                    showCreatorName={true}
                                    showCreatorBadge={true}
                                    parentArgument={parentArgument}
                                    isSupporting={isSupporting}
                                    argument={argument}
                                    key={index}
                                    parentContentCreatorId={parentContentCreatorId}
                                    parentContentType={parentContentType}
                                />
                            )}
                        </tbody>
                    </Table>

                    <div className={"argumentsTableFooter"}>
                        <TableFooter
                            pageNumber={pageNumber}
                            totalPages={totalPages}
                            onPrevious={() => this.handleLoadArguments(pageNumber - 1)}
                            onNext={() => this.handleLoadArguments(pageNumber + 1)}
                        />
                    </div>
                </div>
            );
        }
    }
}

export default withRouter(ArgumentsTable);