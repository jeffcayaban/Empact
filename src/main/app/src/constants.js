export const API_BASE_URL = 'http://localhost:5000/api';
export const ACCESS_TOKEN = 'accessToken';
export const CURRENT_USERNAME = 'username';

export const AUTHORIZATION_HEADER_FIELD = "Authorization";
export const AUTHORIZATION_TOKEN_TYPE = "Bearer";

// Website Name
export const WEBSITE_NAME = "Empact";

// User Roles
export const ROLE_USER = "ROLE_USER";
export const ROLE_ADMIN = "ROLE_ADMIN";

// Argument Types
export const EXPERT_OPINION = "ExpertOpinion";
export const POPULAR_OPINION = "PopularOpinion";

// Argumentation Schemes
export const ARGUMENTATION_SCHEMES = [
    'By Expert Opinion',
    'By Popular Opinion'
];

// Critical Question IDs
export const CRITICAL_QUESTION_IDS = [
    'CQ1',	// Are the circumstances true?
    'CQ2',
    'EO_CQ1',
    'PO_CQ1'
];

// Content Types
export const USER = "User";
export const PETITION = "Petition";
export const ARGUMENT = "Argument";

// Content Properties
export const USERNAME = "Username";
export const ID = "Id";

// Content Actions
export const DELETE = "Delete";
export const CLOSE = "Close";

// Sort Types
export const DESCENDING = "desc";
export const ASCENDING = "asc";
export const SORT_BY_VALUES = [DESCENDING, ASCENDING];

// Content Sort By Values
export const NEWEST = "Newest";
export const OLDEST = "Oldest";
export const MOST_DISCUSSED = "Most Discussed";
export const LEAST_DISCUSSED = "Least Discussed";
export const DISPLAY_SORT_BY_VALUES = [NEWEST, OLDEST, MOST_DISCUSSED, LEAST_DISCUSSED];

// Used in Homepage
export const NO_OF_MOST_DISCUSSED_PETITIONS = 3;
export const NO_OF_MOST_DISCUSSED_ARGUMENTS = 3;

// Petition Form Fields
export const PETITION_TITLE_ID = "title";
export const PETITION_SITUATION_ID = "situation";
export const PETITION_ACTION_ID = "action";
export const PETITION_GOAL_ID = "goal";
export const PETITION_VALUE_ID = "value";
export const PETITION_CLOSING_DATE_ID = "closingDateTime";
export const PETITION_IS_ANONYMOUS_ID = "isAnonymous";

// Register User Fields
export const USERNAME_ID = "username";
export const FIRST_NAME_ID = "firstName";
export const LAST_NAME_ID = "lastName";
export const PASSWORD_ID = "password";
export const CONFIRM_PASSWORD_ID = "confirm-password";

// Register User Field Validation
export const PASSWORD_LENGTH_MIN = 5;
export const PASSWORD_LENGTH_MAX = 21;
export const LAST_NAME_LENGTH_MAX = 40;
export const FIRST_NAME_LENGTH_MAX = 40;
export const USERNAME_LENGTH_MAX = 15;
export const MAXIMUM_NO_OF_SOURCES = 2;
export const MINIMUM_NO_OF_SOURCES = 0;

// Petition Form Validation
export const TITLE_LENGTH_MAX = 90;
export const PETITION_FIELD_MAX = 3000;

// Argument Form Validation
export const EXPERT_FIELD_MAX = 90;
export const ARGUMENT_FIELD_MAX = 6000;

// For Argument Data Visualisation
export const TREE_NETWORK_HEIGHT = 400;

// Other
export const VALID_URL_PREFIX_REGEX = /^(https:\/\/|http:\/\/).*/;
export const PETITION_LIST_SIZE = 5;
export const NOTIFICATION_AUTO_CLOSE_MS = 5000;
export const PETITION_ENTRY_SITUATION_MAX_LENGTH = 400;
export const PETITION_ENTRY_SITUATION_ALT_MAX_LENGTH = 200;
export const MOBILE_MAX_WIDTH = 768;