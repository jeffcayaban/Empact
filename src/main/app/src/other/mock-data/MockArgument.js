import {CRITICAL_QUESTION_IDS, EXPERT_OPINION, ROLE_ADMIN} from "../../constants";

export default {
    id: 'testId',
    isSupporting: true,
    rootPetitionId: 'testRootPetitionId',
    parentArgumentId: 'testParentArgumentId',
    subCriticalQuestionId: null,
    agreesWithCQ: true,
    criticalQuestionId: CRITICAL_QUESTION_IDS[0],
    creationDateTime: 1551716401541,
    lastUpdatedDateTime: 1551716402000,
    createdBy: {
        id: 'testId',
        username: 'testUsername',
        firstName: 'testFirstName',
        lastName: 'testLastName',
        authorities: [ROLE_ADMIN]
    },
    argumentType: EXPERT_OPINION,
    isAnonymous: false,
    sources: [],
    expert: "testExpert",
    expertDomain: 'testDomain',
    expertAssertion: 'testAssertion'
};