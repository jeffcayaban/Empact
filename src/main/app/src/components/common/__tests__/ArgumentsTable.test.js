import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import MockArgument from '../../../other/mock-data/MockArgument';
import MockArguments from '../../../other/mock-data/MockGetArgumentsResponse';
import ArgumentsTable from "../ArgumentsTable";
import * as AppUtils from '../../../utils/AppUtils';

import {ARGUMENT, OLDEST, PETITION} from "../../../constants";

function createArgumentsTable(props = {}) {
    const mockRouter = jest.fn();
    return <ArgumentsTable.WrappedComponent {...props} params={{ router: mockRouter }} />;
}

describe('ArgumentsTable', () => {

    it('should render successfully for an argument whose parent is a petition', () => {
        const originalGetArgumentsByPetition = AppUtils.getArgumentsByRootPetitionId;
        AppUtils.getArgumentsByRootPetitionId = () => Promise.resolve(MockArguments);

        const props = {
            parentArgument: MockArgument,
            isSupporting: true,
            parentContentCreatorId: "testId",
            parentContentType: PETITION
        };

        const component = shallow(createArgumentsTable(props));
        component.setState({ argumentsList: MockArguments.content });

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.getArgumentsByRootPetitionId = originalGetArgumentsByPetition;
    });

    it('should render successfully for the most discussed arguments whose parent is a petition', () => {
        const props = {
            parentArgument: MockArgument,
            isSupporting: true,
            parentContentCreatorId: "testId",
            parentContentType: PETITION
        };

        const component = shallow(createArgumentsTable(props));
        component.setState({ argumentsList: MockArguments.content });

        const instance = component.instance();
        instance.getArgumentsFromServer(0, 1, "asc", "testId", true, PETITION, true);

        expect(toJson(component)).toMatchSnapshot();
    });


    it('should render successfully for an argument whose parent is an argument', () => {
        const originalGetArgumentsByArgumentId = AppUtils.getArgumentsByArgumentId;
        AppUtils.getArgumentsByArgumentId = () => Promise.resolve(MockArguments);

        const props = {
            parentArgument: MockArgument,
            isSupporting: true,
            parentContentCreatorId: "testId",
            parentContentType: ARGUMENT
        };

        const component = shallow(createArgumentsTable(props));
        component.setState({ argumentsList: MockArguments.content });

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.getArgumentsByArgumentId = originalGetArgumentsByArgumentId;
    });

    it('should render successfully for the most discussed arguments whose parent is an argument', () => {
        const props = {
            parentArgument: MockArgument,
            isSupporting: true,
            parentContentCreatorId: "testId",
            parentContentType: ARGUMENT
        };

        const component = shallow(createArgumentsTable(props));
        component.setState({ argumentsList: MockArguments.content });

        const instance = component.instance();
        instance.getArgumentsFromServer(0, 1, "asc", "testId", true, ARGUMENT, true);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully call handleChangeSortBy for Oldest', () => {
        const props = {
            parentArgument: MockArgument,
            isSupporting: true,
            parentContentCreatorId: "testId",
            parentContentType: ARGUMENT
        };

        const component = shallow(createArgumentsTable(props));
        component.setState({ argumentsList: MockArguments.content });

        const instance = component.instance();
        instance.handleChangeSortBy("2");

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully call handleChangeSortBy for Most Discussed', () => {
        const props = {
            parentArgument: MockArgument,
            isSupporting: true,
            parentContentCreatorId: "testId",
            parentContentType: ARGUMENT
        };

        const component = shallow(createArgumentsTable(props));
        component.setState({ argumentsList: MockArguments.content });

        const instance = component.instance();
        instance.handleChangeSortBy("3");

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully call handleChangeSortBy for Least Discussed', () => {
        const props = {
            parentArgument: MockArgument,
            isSupporting: true,
            parentContentCreatorId: "testId",
            parentContentType: ARGUMENT
        };

        const component = shallow(createArgumentsTable(props));
        component.setState({ argumentsList: MockArguments.content });

        const instance = component.instance();
        instance.handleChangeSortBy("4");

        expect(toJson(component)).toMatchSnapshot();
    });
});
