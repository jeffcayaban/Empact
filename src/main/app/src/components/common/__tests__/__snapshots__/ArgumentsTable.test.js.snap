// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArgumentsTable should render successfully call handleChangeSortBy for Least Discussed 1`] = `
<div>
  <div
    className="argumentsTableHeader"
  >
    <div
      className="spaceBetweenElements"
      style={
        Object {
          "paddingBottom": "7px",
          "paddingTop": "7px",
        }
      }
    >
      <TableSortByBtn
        id="true-args-sortby"
        onSelect={[Function]}
        showMoreOptions={true}
        title="Least Discussed"
      />
    </div>
  </div>
  <Table
    bordered={false}
    bsClass="table"
    className="argumentsTable"
    condensed={false}
    hover={true}
    responsive={true}
    striped={false}
  >
    <tbody>
      <withRouter(ArgumentEntry)
        argument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "PopularOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:38Z",
            "criticalQuestionId": "CQ2",
            "explanation": "TestExplanation",
            "id": 1,
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
            "parentArgumentId": null,
            "rootPetitionId": 1,
            "sources": Array [],
          }
        }
        isSupporting={true}
        key="0"
        parentArgument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          }
        }
        parentContentCreatorId="testId"
        parentContentType="Argument"
        showCreatorBadge={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    className="argumentsTableFooter"
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={0}
    />
  </div>
</div>
`;

exports[`ArgumentsTable should render successfully call handleChangeSortBy for Most Discussed 1`] = `
<div>
  <div
    className="argumentsTableHeader"
  >
    <div
      className="spaceBetweenElements"
      style={
        Object {
          "paddingBottom": "7px",
          "paddingTop": "7px",
        }
      }
    >
      <TableSortByBtn
        id="true-args-sortby"
        onSelect={[Function]}
        showMoreOptions={true}
        title="Most Discussed"
      />
    </div>
  </div>
  <Table
    bordered={false}
    bsClass="table"
    className="argumentsTable"
    condensed={false}
    hover={true}
    responsive={true}
    striped={false}
  >
    <tbody>
      <withRouter(ArgumentEntry)
        argument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "PopularOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:38Z",
            "criticalQuestionId": "CQ2",
            "explanation": "TestExplanation",
            "id": 1,
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
            "parentArgumentId": null,
            "rootPetitionId": 1,
            "sources": Array [],
          }
        }
        isSupporting={true}
        key="0"
        parentArgument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          }
        }
        parentContentCreatorId="testId"
        parentContentType="Argument"
        showCreatorBadge={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    className="argumentsTableFooter"
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={0}
    />
  </div>
</div>
`;

exports[`ArgumentsTable should render successfully call handleChangeSortBy for Oldest 1`] = `
<div>
  <div
    className="argumentsTableHeader"
  >
    <div
      className="spaceBetweenElements"
      style={
        Object {
          "paddingBottom": "7px",
          "paddingTop": "7px",
        }
      }
    >
      <TableSortByBtn
        id="true-args-sortby"
        onSelect={[Function]}
        showMoreOptions={true}
        title="Oldest"
      />
    </div>
  </div>
  <Table
    bordered={false}
    bsClass="table"
    className="argumentsTable"
    condensed={false}
    hover={true}
    responsive={true}
    striped={false}
  >
    <tbody>
      <withRouter(ArgumentEntry)
        argument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "PopularOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:38Z",
            "criticalQuestionId": "CQ2",
            "explanation": "TestExplanation",
            "id": 1,
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
            "parentArgumentId": null,
            "rootPetitionId": 1,
            "sources": Array [],
          }
        }
        isSupporting={true}
        key="0"
        parentArgument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          }
        }
        parentContentCreatorId="testId"
        parentContentType="Argument"
        showCreatorBadge={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    className="argumentsTableFooter"
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={0}
    />
  </div>
</div>
`;

exports[`ArgumentsTable should render successfully for an argument whose parent is a petition 1`] = `
<div>
  <div
    className="argumentsTableHeader"
  >
    <div
      className="spaceBetweenElements"
      style={
        Object {
          "paddingBottom": "7px",
          "paddingTop": "7px",
        }
      }
    >
      <TableSortByBtn
        id="true-args-sortby"
        onSelect={[Function]}
        showMoreOptions={true}
        title="Newest"
      />
    </div>
  </div>
  <Table
    bordered={false}
    bsClass="table"
    className="argumentsTable"
    condensed={false}
    hover={true}
    responsive={true}
    striped={false}
  >
    <tbody>
      <withRouter(ArgumentEntry)
        argument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "PopularOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:38Z",
            "criticalQuestionId": "CQ2",
            "explanation": "TestExplanation",
            "id": 1,
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
            "parentArgumentId": null,
            "rootPetitionId": 1,
            "sources": Array [],
          }
        }
        isSupporting={true}
        key="0"
        parentArgument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          }
        }
        parentContentCreatorId="testId"
        parentContentType="Petition"
        showCreatorBadge={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    className="argumentsTableFooter"
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={0}
    />
  </div>
</div>
`;

exports[`ArgumentsTable should render successfully for an argument whose parent is an argument 1`] = `
<div>
  <div
    className="argumentsTableHeader"
  >
    <div
      className="spaceBetweenElements"
      style={
        Object {
          "paddingBottom": "7px",
          "paddingTop": "7px",
        }
      }
    >
      <TableSortByBtn
        id="true-args-sortby"
        onSelect={[Function]}
        showMoreOptions={true}
        title="Newest"
      />
    </div>
  </div>
  <Table
    bordered={false}
    bsClass="table"
    className="argumentsTable"
    condensed={false}
    hover={true}
    responsive={true}
    striped={false}
  >
    <tbody>
      <withRouter(ArgumentEntry)
        argument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "PopularOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:38Z",
            "criticalQuestionId": "CQ2",
            "explanation": "TestExplanation",
            "id": 1,
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
            "parentArgumentId": null,
            "rootPetitionId": 1,
            "sources": Array [],
          }
        }
        isSupporting={true}
        key="0"
        parentArgument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          }
        }
        parentContentCreatorId="testId"
        parentContentType="Argument"
        showCreatorBadge={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    className="argumentsTableFooter"
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={0}
    />
  </div>
</div>
`;

exports[`ArgumentsTable should render successfully for the most discussed arguments whose parent is a petition 1`] = `
<div>
  <div
    className="argumentsTableHeader"
  >
    <div
      className="spaceBetweenElements"
      style={
        Object {
          "paddingBottom": "7px",
          "paddingTop": "7px",
        }
      }
    >
      <TableSortByBtn
        id="true-args-sortby"
        onSelect={[Function]}
        showMoreOptions={true}
        title="Newest"
      />
    </div>
  </div>
  <Table
    bordered={false}
    bsClass="table"
    className="argumentsTable"
    condensed={false}
    hover={true}
    responsive={true}
    striped={false}
  >
    <tbody>
      <withRouter(ArgumentEntry)
        argument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "PopularOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:38Z",
            "criticalQuestionId": "CQ2",
            "explanation": "TestExplanation",
            "id": 1,
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
            "parentArgumentId": null,
            "rootPetitionId": 1,
            "sources": Array [],
          }
        }
        isSupporting={true}
        key="0"
        parentArgument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          }
        }
        parentContentCreatorId="testId"
        parentContentType="Petition"
        showCreatorBadge={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    className="argumentsTableFooter"
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={0}
    />
  </div>
</div>
`;

exports[`ArgumentsTable should render successfully for the most discussed arguments whose parent is an argument 1`] = `
<div>
  <div
    className="argumentsTableHeader"
  >
    <div
      className="spaceBetweenElements"
      style={
        Object {
          "paddingBottom": "7px",
          "paddingTop": "7px",
        }
      }
    >
      <TableSortByBtn
        id="true-args-sortby"
        onSelect={[Function]}
        showMoreOptions={true}
        title="Newest"
      />
    </div>
  </div>
  <Table
    bordered={false}
    bsClass="table"
    className="argumentsTable"
    condensed={false}
    hover={true}
    responsive={true}
    striped={false}
  >
    <tbody>
      <withRouter(ArgumentEntry)
        argument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "PopularOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:38Z",
            "criticalQuestionId": "CQ2",
            "explanation": "TestExplanation",
            "id": 1,
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
            "parentArgumentId": null,
            "rootPetitionId": 1,
            "sources": Array [],
          }
        }
        isSupporting={true}
        key="0"
        parentArgument={
          Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          }
        }
        parentContentCreatorId="testId"
        parentContentType="Argument"
        showCreatorBadge={true}
        showCreatorName={true}
      />
    </tbody>
  </Table>
  <div
    className="argumentsTableFooter"
  >
    <TableFooter
      onNext={[Function]}
      onPrevious={[Function]}
      pageNumber={0}
      totalPages={0}
    />
  </div>
</div>
`;
